### PR TITLE
Refactoring invis interfaces to decouple invis from consume

### PR DESCRIPTION
**Pull request by devshorts**

### CODE REVIEW

**Comment by devshorts** [comment]
I have better ideas of how to refactor this. Will close and re open late

**Comment by devshorts** [comment]
@jakeswenson take a look. I think this will set us up well for the queue based invis strategy

File: core/src/main/java/io/paradoxical/cassieq/workers/DefaultMessageConsumer.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    @Override
+    public Optional<Message> tryConsume(ConsumableMessage message) {
+        if (!messageConsumable(message.getMessage())) {
+            return Optional.empty();
+        }
+
+        final Optional<Message> consumeMessage = messageRepository.rawConsumeMessage(message.getMessage(), message.getInvisibility());
+
+        if (consumeMessage.isPresent()) {
+            invisStrategy.trackConsumedMessage(message.toBuilder().message(consumeMessage.get()).build());
```
Comment (devshorts) [inline core/src/main/java/io/paradoxical/cassieq/workers/DefaultMessageConsumer.java]: This is the delegation to the invis strategy

File: core/src/main/java/io/paradoxical/cassieq/workers/reader/PointerBasedInvisStrategy.java [hunk 2]
```diff
@@ -179,14 +189,7 @@ private InvisMessagePointerProcessResult processCurrentInvisPointer(Invisibility
 
         if (messageAt.isVisible(clock) && messageAt.isNotAcked()) {
             // the message we are pointing at has come back alive
-            final Optional<Message> message = messageConsumer.tryConsume(messageAt, invisibility);
-
-            // were able to consume
-            if (message.isPresent()) {
-                metricRegistry.counter(name("reader", "revived", "messages")).inc();
```
Comment (devshorts) [inline core/src/main/java/io/paradoxical/cassieq/workers/reader/PointerBasedInvisStrategy.java:186]: These metrics are now tracked by the reader

File: core/src/test/java/io/paradoxical/cassieq/unittests/TestBase.java [hunk 3]
```diff
@@ -121,6 +121,7 @@ protected TestQueueContext setupTestContext(String queueName, int bucketSize) {
         final QueueDefinition queueDefinition = QueueDefinition.builder()
                                                                .accountName(testAccountName)
                                                                .queueName(queue)
+                                                               .allowRandomBucketReading(false)
```
Comment (devshorts) [inline core/src/test/java/io/paradoxical/cassieq/unittests/TestBase.java]: Our tests need very speciifc ordering for our asserts to work. I'll make an item to support "hand wavy" asserts that any of a particular range are consumed when we do a "readAndAck"

File: core/src/test/java/io/paradoxical/cassieq/unittests/TestQueueContext.java [hunk 4]
```diff
@@ -118,7 +118,7 @@ public QueueDeleter createQueueDeleter() {
     public boolean readAndAckMessage(String blob, Long invisDuration) {
         Optional<Message> message = getReader().nextMessage(Duration.standardSeconds(invisDuration));
 
-        assertThat(message.get().getBlob().equals(blob));
+        assertThat(message.get().getBlob()).isEqualTo(blob);
```
Comment (devshorts) [inline core/src/test/java/io/paradoxical/cassieq/unittests/TestQueueContext.java:122]: OOPS

File: core/src/main/java/io/paradoxical/cassieq/dataAccess/Tables.java [hunk 5]
```diff
@@ -38,6 +38,7 @@
         public static final String REPAIR_WORKER_TOMBSTONE_BUCKET_TIMEOUT_SECONDS = "repair_worker_tombstone_bucket_timeout_seconds";
         public static final String DELETE_BUCKETS_AFTER_FINALIZATION = "delete_buckets_after_finalization";
         public static final String DLQ_NAME = "dlq_queue_name";
+        public static final String ALLOW_RANDOM_BUCKET_READING = "allow_random_bucket_reading";
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/dataAccess/Tables.java]: what about `strict_fifo` defaulted to true?

Comment (devshorts) [inline core/src/main/java/io/paradoxical/cassieq/dataAccess/Tables.java]: Love it!

File: core/src/main/java/io/paradoxical/cassieq/workers/DefaultMessageConsumer.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * Consumes with business logic
+     *
+     * @param message
+     * @return
+     */
+    @Override
+    public Optional<Message> tryConsume(ConsumableMessage message) {
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/workers/DefaultMessageConsumer.java]: rename parameter to be `consumableMessage`

**Comment by jakeswenson** [comment]
LGTM :heavy_check_mark: super nice to have

**Pull request merged**